### PR TITLE
[v16] Update proxy features

### PR DESCRIPTION
**Pull request by mcbattirola**
Backport #45979 to branch/v16

teleport.e PR: https://github.com/gravitational/teleport.e/pull/5104

### CODE REVIEW

**Comment by mcbattirola** [comment]
> Is there any way to move forward with the backports without breaking the ent builds/tests?

I've opened the teleport.e backport and will only merge this one once the enterprise backport is approved, that should minimize the time the build is broken.

The only change in this PR that breaks the tests/build is renaming `ClusterFeatures` to `clusterFeatures`. I can undo that and, once the teleport.e PR lands, I can open a new PR to make it unexported again. That is more work but won't break the build, is that preferable to breaking the build for a few minutes?

**Review by rosstimothy — APPROVED** [review]
Thanks @mcbattirola!

**Review by kimlisa — APPROVED** [review]

**Pull request merged**
